sdk/core/azure-core/src/main/java/com/azure/core/http/rest/IterableResponse.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import reactor.core.publisher.Flux;
+
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+/**
+ * This class provides utility to iterate over values. All the values are preserved even if they are traversed multiple times.
+ *
+ * <p><strong>Code sample using Stream</strong></p>
+ *
+ * {@codesnippet com.azure.core.http.rest.iterableResponse.stream}
+ *
+ * <p><strong>Code sample using Iterator</strong></p>
+ *
+ * {@codesnippet com.azure.core.http.rest.iterableResponse.iterator.while}
+ *
+ * <p><strong>Code sample using Stream and filter</strong></p>
+ *
+ * {@codesnippet com.azure.core.http.rest.iterableResponse.stream.filter}
+ *
+ * @param <T> value
```

Comment:
The documentation for param seems rather vague. "value". 🤔 Something like "The type of elements in this iterable." tells more.

Reply:
```suggestion
 * @param <T> The type of elements contained in the Iterable.
```